app/javascript/components/layer.jsx

```diff
@@ -24,187 +32,358 @@ export function Layer({ scope, layer, isOwner, layerId, activeTool, activeColor
 
-    if (scope) {
-      scope.view.onMouseDown = (event) => {
-        scope.activate();
-        if (activeTool === 'Pen') {
-          scope.project.deselectAll();
-          pathRef.current = new Paper.Path();
-        } else if (activeTool === 'Eraser') {
-          scope.project.deselectAll();
-          path = new Paper.Path();
-        }
-        else if (activeTool === 'Select') {
-          p1 = new Paper.Point(event.point.x, event.point.y);
-          if (rect && p1.isInside(rect) && scope.project.selectedItems.length > 0) {
-            rect = null;
-            path = null;
-          } else {
-            scope.project.deselectAll();
-            rect = new Paper.Rectangle();
-            path = new Paper.Path.Rectangle(rect);
-          }
-        } else if (activeTool === 'Text') {
+    if (!scope) {
+      return;
+    }
+
+    scope.view.onMouseDown = (event) => {
+      scope.activate();
+      // Transmit text diff if clicking out of a focused text box
+      if (!!pathRef.current && pathRef.current instanceof Paper.PointText) {
+        transmitDiff(DiffType.Tangible, pathRef.current);
+        pathRef.current.fullySelected = false;
+      }
+      // Handle event according to active tool
+      if (activeTool === CanvasTools.pen) {
+        scope.project.deselectAll();
+        pathRef.current = new Paper.Path();
+      } else if (activeTool === CanvasTools.eraser) {
+        scope.project.deselectAll();
+        path = new Paper.Path();
+      } else if (activeTool === CanvasTools.select) {
+        p1 = new Paper.Point(event.point.x, event.point.y);
+        if (rect && p1.isInside(rect) && scope.project.selectedItems.length > 0) {
+          rect = null;
+          path = null;
+        } else {
           scope.project.deselectAll();
-          const point = new Paper.Point(event.point.x, event.point.y);
-          pathRef.current = new Paper.PointText({
-            point: point,
-            fontFamily: 'serif',
-            fontSize: 25,
-          });
-          pathRef.current.fullySelected = true;
+          rect = new Paper.Rectangle();
+          path = new Paper.Path.Rectangle(rect);
         }
-        if (activeTool !== 'Pen' && activeTool !== 'Text' && scope.project.selectedItems.length === 0) {
+      } else if (activeTool === CanvasTools.text) {
+        scope.project.deselectAll();
+        const point = new Paper.Point(event.point.x, event.point.y);
+        pathRef.current = new Paper.PointText({
+          point: point,
+          fontFamily: 'serif',
+          fontSize: 25,
+        });
+        pathRef.current.fullySelected = true;
+      }
+
+      if (scope.project.selectedItems.length === 0) {
+        if (activeTool === CanvasTools.eraser || activeTool === CanvasTools.select) {
           path.strokeColor = 'black';
           path.strokeWidth = 3;
           path.dashArray = [10, 12];
-        } else if (scope.project.selectedItems.length === 0 || (activeTool === 'Text')) {
+        } else {
```

Comment:
Because L317 sets a PointText to selected upon PointText creation, the check on L320 will return false for text diffs. This prevents the user from setting the color of their text diffs.

Reply:
Oh ya good catch on this. Will change rn.